lib/buffer.js

```diff
@@ -590,11 +589,8 @@ Buffer.prototype.toString = function(encoding, start, end) {
 
     if (end <= start)
```

Comment:
You added an early `return` which eliminates a correct (but very tricky) behavior in the old code where it would check if `end <= start` for undefined values. I'm just saying I'm happy with the change :)